cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaVersion.java

```diff
@@ -135,43 +136,145 @@ public KafkaVersion version(String version) {
+            }
+        }
+
+        /**
+         * The Kafka Connect image to use for a Kafka Connect cluster.
+         * @param image The image given in the CR.
+         * @param version The version given in the CR.
+         * @return The image to use.
+         * @throws InvalidResourceException If no image was given in the CR and the version given
+         * was not present in {@link ClusterOperatorConfig#STRIMZI_KAFKA_CONNECT_IMAGES}.
+         */
         public String kafkaConnectVersion(String image, String version) {
-            return image(image,
-                    version,
-                    kafkaConnectImages);
+            try {
+                return image(image,
+                        version,
+                        kafkaConnectImages,
+                        ClusterOperatorConfig.STRIMZI_KAFKA_CONNECT_IMAGES);
+            } catch (NoImageException e) {
+                throw new InvalidResourceException(e);
+            }
+        }
+
+        /**
+         * Validate that the given versions have images present in {@link ClusterOperatorConfig#STRIMZI_KAFKA_CONNECT_IMAGES}.
+         * @param versions The versions to validate.
+         * @throws NoImageException If one of the versions lacks an image.
+         */
+        public void validateKafkaConnectImages(Iterable<String> versions) throws NoImageException {
+            for (String version : versions) {
+                image(null, version, kafkaConnectImages, ClusterOperatorConfig.STRIMZI_KAFKA_CONNECT_IMAGES);
+            }
         }
 
+        /**
+         * The Kafka Connect S2I image to use for a Kafka Connect cluster.
+         * @param image The image given in the CR.
+         * @param version The version given in the CR.
+         * @return The image to use.
+         * @throws InvalidResourceException If no image was given in the CR and the version given
+         * was not present in {@link ClusterOperatorConfig#STRIMZI_KAFKA_CONNECT_S2I_IMAGES}.
+         */
         public String kafkaConnectS2iVersion(String image, String version) {
-            return image(image,
-                    version,
-                    kafkaConnectS2iImages);
+            try {
+                return image(image,
+                        version,
+                        kafkaConnectS2iImages,
+                        ClusterOperatorConfig.STRIMZI_KAFKA_CONNECT_S2I_IMAGES);
+            } catch (NoImageException e) {
+                throw new InvalidResourceException(e);
+            }
+        }
+
+        /**
+         * Validate that the given versions have images present in {@link ClusterOperatorConfig#STRIMZI_KAFKA_CONNECT_S2I_IMAGES}.
+         * @param versions The versions to validate.
+         * @throws NoImageException If one of the versions lacks an image.
+         */
+        public void validateKafkaConnectS2iImages(Iterable<String> versions) throws NoImageException {
```

Comment:
We seem to use `S2I` in names of other classes / methods. Should we follow it here as well?